Frameworks/CoreText/CTFramesetter.mm

```diff
@@ -36,49 +36,71 @@ static id _createFrame(_CTFrameSetter* frameSetter, CGRect frameSize, CGSize* si
         ret->_lines.attach([NSMutableArray new]);
     }
 
-    sizeOut->width = 0;
-    sizeOut->height = 0;
-
-    float y = frameSize.size.height; //[font ascender];
-    CFIndex curIdx = 0;
-    NSString* string = [frameSetter->_typesetter->_attributedString string];
-    CFIndex stringRange = [string length];
-
-    for (;;) {
-        CFIndex pos = CTTypesetterSuggestLineBreakWithOffset((CTTypesetterRef)(_CTTypesetter*)frameSetter->_typesetter,
-                                                             curIdx,
-                                                             frameSize.size.width,
-                                                             0.0f);
-        if (pos == curIdx)
-            break;
-
-        CFRange lineRange;
-        lineRange.location = curIdx;
-        lineRange.length = pos - curIdx;
-
-        CTLineRef line = CTTypesetterCreateLineWithOffset((CTTypesetterRef)(_CTTypesetter*)frameSetter->_typesetter, lineRange, 0.0f);
-
-        float ascent = 0.0f, descent = 0.0f, leading = 0.0f;
-        float width = CTLineGetTypographicBounds(line, &ascent, &descent, &leading);
-        float lineHeight = ascent - descent + leading;
-
-        if (ret) {
-            CGPoint lineOrigin;
-            lineOrigin.x = 0.0f;
-            lineOrigin.y = y - ascent;
-            [ret->_lines addObject:(id)line];
-            ret->_lineOrigins.push_back(lineOrigin);
+    // Only fill in frame if there is text
+    if (CFAttributedStringGetLength(reinterpret_cast<CFAttributedStringRef>(frameSetter->_typesetter->_attributedString))) {
```

Comment:
Date: Fri, 12 Aug 2016 10:54:46 -0700
From: notifications@github.com
To: WinObjC@noreply.github.com
Subject: Re: [Microsoft/WinObjC] Handle core text right and center alignment (#873)

In Frameworks/CoreText/CTFramesetter.mm:cvfffgsederrf  nn


> -        lineRange.length = pos - curIdx;
> -
> -        CTLineRef line = CTTypesetterCreateLineWithOffset((CTTypesetterRef)(_CTTypesetter*)frameSetter->_typesetter, lineRange, 0.0f);
> -
> -        float ascent = 0.0f, descent = 0.0f, leading = 0.0f;
> -        float width = CTLineGetTypographicBounds(line, &ascent, &descent, &leading);
> -        float lineHeight = ascent - descent + leading;
> -
> -        if (ret) {
> -            CGPoint lineOrigin;
> -            lineOrigin.x = 0.0f;
> -            lineOrigin.y = y - ascent;
> -            [ret->_lines addObject:(id)line];
> -            ret->_lineOrigins.push_back(lineOrigin);
> +    // Only fill in frame if there is text
> +    if (CFAttributedStringGetLength(reinterpret_cast<CFAttributedStringRef>(frameSetter->_typesetter->_attributedString))) {


reinterpret_cast 



Use static_cast here, reinterpret is quite a bit stronger than needed for this.


—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or mute the thread.